diff --git a/node_modules/@semantic-release/release-notes-generator/index.js b/node_modules/@semantic-release/release-notes-generator/index.js
index 8cbd5f8..111d9fa 100644
--- a/node_modules/@semantic-release/release-notes-generator/index.js
+++ b/node_modules/@semantic-release/release-notes-generator/index.js
@@ -55,6 +55,10 @@ async function generateNotes(pluginConfig, context) {
         ...rawCommit,
         ...parser(rawCommit.message, {referenceActions, issuePrefixes, ...parserOpts}),
       }))
+      .map(parsedCommit => ({
+        ...parsedCommit,
+        notes: parsedCommit.notes.map(fixCommitNote),
+      }))
   );
   const previousTag = lastRelease.gitTag || lastRelease.gitHead;
   const currentTag = nextRelease.gitTag || nextRelease.gitHead;
@@ -86,7 +90,20 @@ async function generateNotes(pluginConfig, context) {
   debug('issue: %o', changelogContext.issue);
   debug('commit: %o', changelogContext.commit);
 
-  return getStream(intoStream.object(parsedCommits).pipe(writer(changelogContext, writerOpts)));
+  const result = getStream(intoStream.object(parsedCommits).pipe(writer(changelogContext, writerOpts)))
+
+  return result;
 }
 
 module.exports = {generateNotes};
+
+function fixCommitNote({ title, text }) {
+  text = text
+    .replace(/signed-off-by:.+(\n|$)/i, '')
+    .replace(/(^|\n+)-\s+/g, '--newline--• ')
+    .replace(/\n/g, ' ')
+    .replace(/--newline--/g, '\n')
+    .replace(/ +/g, ' ');
+
+  return { title, text };
+}
\ No newline at end of file
